kubernetes/cray-nexus/templates/hooks.yaml

```diff
@@ -1,3 +1,66 @@
+#
+# MIT License
+#
+# (C) Copyright 2020-2024 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+{{- if .Values.security.keycloak.enabled }}
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ template "cray-nexus.fullname" . }}-secret-check
+  namespace: {{ .Release.Namespace }}
+  labels:
+{{ include "cray-nexus.labels" . | indent 4 }}
+  annotations:
+    helm.sh/hook: pre-install, pre-upgrade
+    helm.sh/hook-delete-policy: before-hook-creation, hook-succeeded
+spec:
+  template:
+    metadata:
+      annotations:
+        sidecar.istio.io/inject: "false"
+    spec:
+      serviceAccountName: "{{ .Values.hooks.preInstall.serviceAccountName }}"
+      restartPolicy: Never
+      containers:
+        - name: secret-check
+          image: "{{ .Values.hooks.preInstall.image.repository }}:{{ .Values.hooks.preInstall.image.tag }}"
+          imagePullPolicy: "{{ .Values.hooks.preInstall.image.pullPolicy }}"
+          command:
+          - /bin/sh
+          - -c
+          - |-
+            set -x
+
+            echo "Looking for secret {{ .Values.security.keycloak.client.secret.name }} ..."
+            # Wait for secret to exist in nexus namespace
+            while [[ ! $(kubectl get secrets -n {{ .Release.Namespace }} | awk '{print $1}' | grep {{ .Values.security.keycloak.client.secret.name }}) ]]; do
```

Comment:
Is there a reason this can't just be:
```
while ! kubectl get secrets -n {{ .Release.Namespace }} {{ .Values.security.keycloak.client.secret.name }} >/dev/null; do
```

Reply:
Nope.  There is no reason it can't be that instead.  I'll change it.